src/server/game/Movement/Spline/MoveSplineInit.h

```diff
@@ -149,7 +149,7 @@ namespace Movement
         /* Sets the velocity (in case you want to have custom movement velocity)
          * if no set, speed will be selected based on unit's speeds and current movement mode
          * Has no effect if falling mode enabled
-         * velocity shouldn't be negative
+         * velocity shouldn't be negative (except knockbacks)
```

Comment:
Is there any other place in spline code left that does not force abs(velocity)?

Reply:
I've avoided modifying anything that wouldn't affect knockbacks or that I didn't know what it is used for. Checking what you comment:
- https://github.com/TrinityCore/TrinityCore/blob/master/src/server/game/Movement/Spline/MoveSpline.cpp#L118 (unused?)
- In CommonInitializer there is no check and I've only changed the parameter passed in its call https://github.com/TrinityCore/TrinityCore/blob/1fc3a51db2d7248c62e0890ae1e0b40f83d9ae2c/src/server/game/Movement/Spline/MoveSpline.cpp#L168
- In MoveSpline I don't know if you prefer to save the velocity in absolute value https://github.com/TrinityCore/TrinityCore/blob/master/src/server/game/Movement/Spline/MoveSpline.cpp#L181
- MotionMaster::MoveJump and MotionMaster::MoveJumpWithGravity are checking that speedXY is always positive (which I understand is correct)